### PR TITLE
Move to newer CodeBlock API for emitting new lines

### PR DESCRIPTION
**Pull request by shaishavgandhi05**
Builds on the work done by #568 and uses the same API for `FunSpec`. It's a little trickier and tedious here because of the different types of kdoc tags and the need for adding a breakline before them in case that's where the tags begin.

### CODE REVIEW

File: src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt [hunk 1]
```diff
@@ -259,6 +259,22 @@ class FunSpecTest {
...
+        .addCode("return %S", "foo")
+        .addKdoc("Do the foo")
+        .build()
+    assertThat(funSpec.toString()).isEqualTo("""
+      |/**
+      | * Do the foo
+      | *
+      | * @return the foo.
+      | */
+      |fun foo(nodoc: kotlin.Boolean): kotlin.String = "foo"""".trimMargin())
```
Comment (oldergod) [inline src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: Don't we want a new line before `"""`?

Comment (swankjesse) [reply src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: Yes please

Comment (Egorand) [reply src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: Ouch, this snippet actually looks broken, notice the extra closing bracket at the end of the line... Normally you'd do `.addStatement("return %S", "foo")`, which would emit correct code. Not sure this is worth filing.

Comment (shaishavgandhi05) [reply src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: Thanks! I went around and fixed that for some other tests also (written by yours truly, guilty as charged). 
0b9f23257bfa0ec49dac663c93b43ab3d128c83b

File: src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt [hunk 2]
```diff
@@ -516,6 +532,24 @@ class FunSpecTest {
       """.trimMargin())
   }
 
+  @Test fun receiverWithKdocAndMainKDoc() {
+    val funSpec = FunSpec.builder("toBar")
+        .receiver(String::class, kdoc = "the string to transform.")
+        .returns(String::class)
+        .addKdoc("%L", "Converts to bar")
+        .addCode(CodeBlock.of("return %S", "bar"))
```
[the remark this answers is not in the document]
Comment (shaishavgandhi05) [reply src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: 0b9f23257bfa0ec49dac663c93b43ab3d128c83b

**Pull request merged**